src/code-util.jl

```diff
@@ -281,13 +335,23 @@ function extract_ics(results)
         return ics
     end
 
-    # Diagnostic categories have identical ordering so we can use row to find matches
-    # across categories, starting with row 1
     for i in 1:length(results[IC_LINE_KEY][1])
         line = extract_detail(results, IC_LINE_KEY, 2, i)
         report = extract_detail(results, IC_REPORT_KEY, 2, i)
 
-        ic = (; line, report)
+        # IC Diagnostic values are indexed by hash and type so we extract them separately
+        h = extract_detail(results, IC_LINE_KEY, 1, i)
+        vs = filter_ic_results(results, IC_OUTPUT_KEY, h)
+        if length(vs) > 11
+            vs = vcat(vs[1:9], "...", last(vs))
+        end
```

Comment:
should we pass in the max number of rows to the filter so we can early exit the construction?

Reply:
Done. I also changed the output to be "value, value, value, ..." instead of "value, value, ..., value" since there is no claim of ordering, thus no 'end' to show. (That, and the code is easier to read).